sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -460,7 +460,12 @@ abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
               // because `distinctExpressions` is not extracted during logical phase.
               NormalizeFloatingNumbers.normalize(e) match {
                 case ne: NamedExpression => ne
-                case other => Alias(other, other.toString)()
+                case other =>
+                  val name = e match {
+                    case ne: NamedExpression => ne.name
+                    case other => other.toString
```

Comment:
This `other.toString` is correct instead of `other` in line 463?

Reply:
Yes, that `other` is `KnownFloatingPointNormalized(NormalizeNaNAndZero(expr))`, which has a very long string representation.